File: 0601-0700/0697-Degree of an Array/0697-Degree of an Array.cpp
class Solution {
public:
    int findShortestSubArray(vector<int>& nums) {
        int freq = 0;
        unordered_map<int, int> start;
        unordered_map<int, int> count;
        int len = nums.size();
        for (int i = 0; i < nums.size(); ++i) {
            if (start.find(nums[i]) == start.end()) {
                start[nums[i]] = i;
            }
            ++count[nums[i]];
            if (count[nums[i]] > freq) {
                freq = count[nums[i]];
                len = i - start[nums[i]] + 1;
            }
            else if (count[nums[i]] == freq) {
                len = min(len, i - start[nums[i]] + 1);
            }            
        }
        
        return len;
    }
};

File: 0601-0700/0697-Degree of an Array/0697-Degree of an Array.py
import collections
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        degree = minLen = 0
        start = {}
        count = collections.Counter()
        for i, num in enumerate(nums):
            if num not in start:
                start[num] = i
            count[num] += 1
            if count[num] > degree:
                degree = count[num]
                minLen = i - start[num] + 1
            elif count[num] == degree:
                minLen = min(minLen, i - start[num] + 1)
        return minLen
